src/software/ai/hl/stp/action/intercept_ball_action.cpp

```diff
@@ -121,7 +121,7 @@ void InterceptBallAction::calculateNextIntent(IntentCoroutine::push_type& yield)
                 DribblerEnable::ON, MoveType::NORMAL, AutokickType::NONE,
                 BallCollisionType::ALLOW));
         }
-    } while (!Evaluation::robotHasPossession(ball, *robot) || loop_forever);
+    } while (!Evaluation::robotHasPossession(ball, *robot));
```

Comment:
I don't think we should still have a while loop here since the loop_forever is in the superclass

Reply:
If im understanding it correctly, I think the loop forever would kick in when this while loop terminates, so we may still need this loop

Reply:
@akhilveeraghanta you mean the `loop_forever`?

Reply:
I mean the loop itself, we can’t remove it entirely.

Reply:
yeah so for cases where we instantiate this action with `loop_forever = false `, we still want the action to continue until we get the ball, but that logic shouldnt be moved to the superclass as its specific to this action